### PR TITLE
Feat/failed reasons

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Create ui for failed reasons page
- Separate logic into custom hook
- Use texts with i18n approach

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-144)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-12-11_15-27-27
photo_2024-12-11_15-28-09
photo_2024-12-11_15-28-13

### CODE REVIEW

File: src/pages/FailedReason/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { Box } from '@mui/system';
+import { t } from 'i18next';
+import React from 'react';
```
Comment (Parle648) [inline src/pages/FailedReason/index.tsx]: react to 1 line

File: src/pages/FailedReason/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  ButtonContainer,
+  Container,
+  InputField,
+  ReasonButton,
+  Subtitle,
+  Title,
+} from './styles';
+import useFailedReason from './useFailedReason';
+
+import Button from '@/components/Button';
```
Comment (Parle648) [inline src/pages/FailedReason/index.tsx:17]: 1.1.14

File: src/pages/FailedReason/styles.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  [theme.breakpoints.up('sm')]: {
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    gap: '16px',
+    justifyContent: 'center',
+    marginBottom: '40px',
+  },
+}));
+
+export const ReasonButton = styled(Button)(({ theme }) => ({
```
Comment (Parle648) [inline src/pages/FailedReason/styles.ts]: types for theme?

File: src/pages/FailedReason/useFailedReason.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { t } from 'i18next';
+import { useState } from 'react';
```
Comment (Parle648) [inline src/pages/FailedReason/useFailedReason.ts:1]: same

File: src/pages/FailedReason/useFailedReason.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+const useFailedReason = (): UseFailedReasonReturn => {
+  const tBase = 'failReason';
+  const minLetters = 10;
+  const maxLetters = 5000;
+
+  const reasonNotAvailable = t(`${tBase}.reasons.notAvailable`);
+  const reasonBagsNotReady = t(`${tBase}.reasons.bagsNotReady`);
+  const reasonOther = t(`${tBase}.reasons.other`);
+
```
Comment (Parle648) [inline src/pages/FailedReason/useFailedReason.ts:12]: move to constants

File: src/pages/FailedReason/useFailedReason.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  const [selectedReason, setSelectedReason] = useState<string | null>(null);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useForm({
```
Comment (Parle648) [inline src/pages/FailedReason/useFailedReason.ts]: useForm<>() create interface for it

File: src/routes/router.tsx [hunk 7]
```diff
@@ -5,6 +5,7 @@ import ProtectedRoute from './ProtectedRoute';
 import RestrictedRoute from './RestrictedRoute';
```
Comment (Parle648) [inline src/routes/router.tsx:5]: 1.1.14

**Review by AlexeyBezrebriy — APPROVED** [review]

**Review by Parle648 — APPROVED** [review]

**Pull request merged**